packages/core/src/parser/util.ts

```diff
@@ -560,3 +560,14 @@ export function acceptIf<P extends Parser<AstNode>>(
 		return ans
 	}) as ReturnType<typeof acceptIf<P>>
 }
+
+/**
+ * @returns A parser that dumps any parser errors after it finishes parsing.
+ */
+export function dumpErrors<P extends Parser<AstNode>>(parser: P): P {
+	return ((src: Source, ctx: ParserContext) => {
+		const ans = parser(src, ctx)
+		ctx.err.dump()
+		return ans
+	}) as ReturnType<typeof dumpErrors<P>>
+}
```

Comment:
this makes sense and LGTM. i'll add a commit to this PR with unit tests for this new util parser tonight

Reply:
done in [`50cfd67` (#1132)](https://github.com/SpyglassMC/Spyglass/pull/1132/commits/50cfd676d186581d95b9447c8757e295a4fc38df)